### PR TITLE
DS18B20sht

### PR DESCRIPTION
**Pull request by Mjohncecil**
Allows the esp8266 to use data from the waterproof thermometer when the sht31 is unavailable.

Test:
If the esp8266 has not been installed in Arduino IDE do so now. (https://randomnerdtutorials.com/how-to-install-esp8266-board-arduino-ide/)
Upload sketch to esp8266.
Connect the sht31, relay, and DS18B20 to esp8266.

sht31: VIN > 3V3, GND > GND, SCL > D1, SDA > D2.
Relay: GND > GND, IN1 > D4, IN2 > D5, VCC > Vin.
DS18B20: follow image at, https://randomnerdtutorials.com/esp8266-ds18b20-temperature-sensor-web-server-with-arduino-ide/ but put output at pin D3 instead of D2.

Run sketch on the esp8266, and pull up the serial monitor.
Check that both temperature and humidity is being read to confirms sht is connected.
Disconnect sht31 by unplugging the SDA cable.
Check the serial monitor to confirm that the esp8266 is still reading the temperature from the DS18B20.

To confirm that the DS18B20 can interact with the relay... 
Keep sht disconnected.
Make note of the current temperature in the room, and edit the "tempLimit" variable in the sketch to be a few degrees above it. 
Re-upload the sketch to the board.
Pull up the serial monitor and confirm that the temperature relay is lit up.
Hold the thermometer in your hand to warm it up above the current "tempLimit"
If the relay light switches off after the recorder temperature exceeds the limit everthing is working correctly.

### CODE REVIEW

**Comment by donnyhyun** [comment]
Looks good! Thanks for walking through the code with me

File: ESP_SHT3x/ESP_SHT3x.ino [hunk 1]
```diff
@@ -1,21 +1,95 @@
...
 #include <Arduino.h>
 #include <Wire.h>
+#include <DallasTemperature.h>
 #include "Adafruit_SHT31.h"
+
+//const char* ssid = "TP-Link_4346";
+//const char* password = "84850034";
+
+const char* ssid = "inHouseTest1";
+const char* password = "nasturtium";
```
Comment (ocasta181) [inline ESP_SHT3x/ESP_SHT3x.ino]: again, do not commit creds. please remove :)

File: ESP_SHT3x/ESP_SHT3x.ino [hunk 2]
```diff
@@ -32,14 +106,50 @@ void printShtMeasurments(float temp, float hum) {
...
-  shtInit();
+
+// Sets relay pins to high or low depending on measurements
+void relayPower(int pin,float current, float limit) {
+  if(current < limit) {
+    digitalWrite(pin, LOW);
+  } else {
+    digitalWrite(pin, HIGH);
+  }
+}
```
Comment (ocasta181) [inline ESP_SHT3x/ESP_SHT3x.ino]: nice abstraction

File: ESP_SHT3x/ESP_SHT3x.ino [hunk 3]
```diff
@@ -32,14 +106,50 @@ void printShtMeasurments(float temp, float hum) {
...
+  // Sets relay relevent pins to appropriate voltage levels
+  relayPower(humidPin, h, humidLimit);
+  relayPower(tempPin, t, tempLimit);
+}
+
+void DS18B20Algorithm() {
+  sensors.requestTemperatures(); 
+  float temperatureC = sensors.getTempCByIndex(0);
+
+  relayPower(tempPin, temperatureC, tempLimit);
```
Comment (ocasta181) [inline ESP_SHT3x/ESP_SHT3x.ino:145]: in addition to setting the heater on/off based on the readings, we'll want to set the humidifier based on a timer.

File: ESP_SHT3x/ESP_SHT3x.ino [hunk 4]
```diff
@@ -32,14 +106,50 @@ void printShtMeasurments(float temp, float hum) {
...
+  
+  Serial.print(temperatureC);
+  Serial.println("ºC");
+}
+ 
+void loop() {
+  if(sht31.begin(0x44)) {
+    shtAlgorithm();
+  }else {
```
Comment (ocasta181) [inline ESP_SHT3x/ESP_SHT3x.ino]: spacing

Comment (ocasta181) [reply ESP_SHT3x/ESP_SHT3x.ino]: also, we'll want to make this an `else if` and include a check to ensure the backup thermometer is accessible.

In the case that neither the SHT31 or the DS18B20 are working, we should fall back to a timer based system (once an hour, on for one minute)

**Review by ocasta181 — CHANGES_REQUESTED** [review]
please review my comments

File: ESP_SHT3x/ESP_SHT3x.ino (hunk 3 above)
Comment (Mjohncecil) [reply ESP_SHT3x/ESP_SHT3x.ino:145]: moved to a new sprint

**Pull request merged**